hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/AbstractClientInternalCacheProxy.java

```diff
@@ -298,8 +298,22 @@ protected void onRemoveAsyncInternal(Data keyData, ClientDelegatingFuture future
         addCallback(future, callback);
     }
 
-    protected <T> ICompletableFuture<T> replaceInternal(K key, V oldValue, V newValue, ExpiryPolicy expiryPolicy,
-                                                        boolean hasOldValue, boolean withCompletionEvent, boolean async) {
+    protected boolean replaceSyncInternal(K key, V oldValue, V newValue, ExpiryPolicy expiryPolicy, boolean hasOldValue) {
+        long startNanos = nowInNanosOrDefault();
+        Future<Boolean> future = replaceAsyncInternal(key, oldValue, newValue, expiryPolicy, hasOldValue, true, false);
+        try {
+            boolean replaced = future.get();
+            if (statisticsEnabled) {
+                statsHandler.onReplace(false, startNanos, replaced);
+            }
+            return replaced;
+        } catch (Throwable e) {
+            throw rethrowAllowedTypeFirst(e, CacheException.class);
+        }
+    }
+
+    protected <T> ICompletableFuture<T> replaceAsyncInternal(K key, V oldValue, V newValue, ExpiryPolicy expiryPolicy,
```

Comment:
This method was asynchronous before, since it always returns a future. The parameter `async` just determined if the statistics should be updated asynchronously. So I changed the method name to have a good counterpart to `replaceSyncInternal()`.